policy_parser.py

```diff
@@ -14,35 +14,107 @@ def __init__(self, msg=None, policy=None):
 
 
 class PolicyParser(object):
+    """
+    Parser for Kubernetes NetworkPolicy API objects.
+
+    They are defined by the NetworkPolicySpec in the /apis/networking API
+    group.
+    """
     def __init__(self, policy):
-        self.namespace = policy["metadata"]["namespace"]
+        """
+        Create a Parser for a Kubernetes NetworkPolicy API object.
+
+        Returns the parser.
+        Throws PolicyError if the Policy is invalid.
```

Comment:
I'll just rip the throws out.  It made more sense before (it was a KeyError + a nicer message).